core/src/main/java/org/hibernate/ogm/jpa/impl/OgmStoredProcedureQuery.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.jpa.impl;
+
+import java.util.List;
+import java.util.Set;
+
+import javax.persistence.EntityManager;
+
+import org.hibernate.HibernateException;
+import org.hibernate.jpa.internal.StoredProcedureQueryImpl;
+import org.hibernate.jpa.spi.HibernateEntityManagerImplementor;
+import org.hibernate.ogm.storedprocedure.impl.NoSQLProcedureResultSetOutputImpl;
+import org.hibernate.procedure.ParameterRegistration;
+import org.hibernate.procedure.ProcedureCall;
+import org.hibernate.procedure.ProcedureCallMemento;
+import org.hibernate.result.ResultSetOutput;
+
+/**
+ * @author Davide D'Alto
+ * @author Sergey Chernolyas &amp;sergey_chernolyas@gmail.com&amp;
+ */
+public class OgmStoredProcedureQuery extends StoredProcedureQueryImpl {
+
+	public OgmStoredProcedureQuery(ProcedureCall procedureCall, EntityManager entityManager) {
+		super( procedureCall, convert( entityManager ) );
+	}
+
+	public OgmStoredProcedureQuery(ProcedureCallMemento procedureCallMemento, EntityManager entityManager) {
+		super( procedureCallMemento, convert( entityManager ) );
+	}
+
+	private static HibernateEntityManagerImplementor convert(EntityManager em) {
+		return em.unwrap( HibernateEntityManagerImplementor.class );
+	}
+
+	@Override
+	public List getResultList() {
+		try {
+			final NoSQLProcedureResultSetOutputImpl rtn = (NoSQLProcedureResultSetOutputImpl) outputs().getCurrent();
```

Comment:
Do we really need this class? Apart from this cast, which is quite artificial and doesn't look like required, it seems to be very generic?

Is there something specific forcing us to not use an ORM class?